pkg/sql/parser/eval.go

```diff
@@ -2088,6 +2088,11 @@ func (expr *FuncExpr) Eval(ctx *EvalContext) (Datum, error) {
 
 	res, err := expr.fn.fn(ctx, args)
 	if err != nil {
+		// If we are facing a retry error, propagate it unchanged, so that
+		// the executor can see it with the right type.
+		if _, ok := err.(*roachpb.RetryableTxnError); ok {
```

Comment:
Hmm, might not be worth it then. But change the comment above the special case for `RetryableTxnError` to explain specifically that this is necessary to support `crdb_internal.force_retry`.